Replace language sidebar with dedicated language screen

xmrdsc opened the pull request:
Due to a hard to debug buildbot issue concerning the sidebar on a statically compiled Qt 5.7.0, I concluded the pragmatic solution is to temporarily disable the sidebar for now until we migrate to > Qt 5.9.

![](https://i.imgur.com/8fxzgnk.png)

xiphon commented on wizard/WizardLang.qml:
```diff
@@ -0,0 +1,236 @@
+        anchors.top: parent.top
+        anchors.topMargin: persistentSettings.customDecorations ? 90 * scaleRatio : 32 * scaleRatio
+        Layout.alignment: Qt.AlignHCenter;
+        width: parent.width - 100
```
`scaleRatio`

xiphon commented on wizard/WizardLang.qml:
```diff
@@ -0,0 +1,236 @@
+    }
+
+    Image {
+        opacity: 1.0
```
```suggestion
```

xiphon commented on wizard/WizardLang.qml:
```diff
@@ -0,0 +1,236 @@
+            Layout.fillWidth: true
+            Layout.topMargin: 20 * scaleRatio
+
+            spacing: 5
```
scaleRatio

xiphon commented on wizard/WizardLang.qml:
```diff
@@ -0,0 +1,236 @@
+                            source: flag
+                            mipmap: true
+                            smooth: true
+                            width: 32
```
scaleRatio

xiphon commented on wizard/WizardLang.qml:
```diff
@@ -0,0 +1,236 @@
+                        anchors.left: img.right
+                        color: "transparent"
+                        height: parent.height
+                        width: langText.width + 22
```
scaleRatio

xiphon commented on wizard/WizardLang.qml:
```diff
@@ -0,0 +1,236 @@
+        anchors.topMargin: persistentSettings.customDecorations ? 90 * scaleRatio : 32 * scaleRatio
+        Layout.alignment: Qt.AlignHCenter;
+        width: parent.width - 100
+        Layout.fillWidth: true
```
```suggestion
```

xiphon commented on wizard/WizardLang.qml:
```diff
@@ -0,0 +1,236 @@
+    ColumnLayout {
+        anchors.top: parent.top
+        anchors.topMargin: persistentSettings.customDecorations ? 90 * scaleRatio : 32 * scaleRatio
+        Layout.alignment: Qt.AlignHCenter;
```
```suggestion
        anchors.horizontalCenter: parent.horizontalCenter
```

xiphon commented on wizard/WizardLang.qml:
```diff
@@ -0,0 +1,236 @@
+
+        TextArea {
+            Layout.fillWidth: true
+            anchors.horizontalCenter: parent.horizontalCenter
```
```suggestion
            Layout.alignment: Qt.AlignHCenter;
```

xiphon commented:
Would be better if the flags were aligned. Might be worth it to set `Repeater -> delegate: Rectangle -> width` to some fixes value

erciccione commented:
> Would be better if the flags were aligned

I agree, but if it's just temporary for this release i don't mind much

mmbyday commented on wizard/WizardLang.qml:
```diff
@@ -0,0 +1,236 @@
+                            wizard.language_wallet = wallet_language;
+                            wizard.language_language = display_name + " (" + locale_spl[1] + ") ";
+
+                            appWindow.showStatusMessage(qsTr("Language changed."), 3);
```
```suggestion
                            appWindow.showStatusMessage(qsTr("Language changed."), 3);
                            appWindow.toggleLanguageView();
```

Better user experience.

xmrdsc replied:
Good catch.

mmbyday commented on wizard/WizardHome.qml:
```diff
@@ -153,9 +153,7 @@ Rectangle {
-                        wizardController.skipModeSelection = true;
-                        wizardController.wizardState = 'wizardLanguage';
-                        languageSidebar.open();
+                        appWindow.toggleLanguageView();
```
The settings > layout > change language button needs updating too.

xmrdsc replied:
Thanks, looks like I did not commit this file.

xmrdsc replied:
Thanks

xmrdsc commented:
> Would be better if the flags were aligned. Might be worth it to set `Repeater -> delegate: Rectangle -> width` to some fixes value

![](https://i.imgur.com/x2ZyHov.png)

> Scaleratios

Fixed

> The settings > layout > change language button needs updating too.

Fixed
